src/doc-printer.js

```diff
@@ -3,6 +3,33 @@
 const MODE_BREAK = 1;
 const MODE_FLAT = 2;
 
+function rootIndent() {
+  return {
+    indent: 0,
+    align: {
+      spaces: 0,
+      tabs: 0
+    }
+  };
+}
+
+function indent(ind) {
+  return {
+    indent: ind.indent + 1,
+    align: ind.align
+  };
+}
+
+function align(ind, n) {
+  return {
+    indent: ind.indent,
+    align: {
+      spaces: ind.align.spaces + n,
+      tabs: ind.align.tabs + (n ? 1 : 0)
```

Comment:
@vjeux :

> Question to you, if you have nested ternaries and other indentation, you're going to have `<tab><space><tab><space>`. Isn't it an issue in practice?

When printing code indented with tabs, my current solution replaces all calls to `align()` with exactly one tab - doesn't matter how many `spaces` it was aligned with, unless it was zero, then it prints zero tabs also.